### ISSUE TITLE
Need proxy support for providers

### ISSUE THREAD

**Issue by fakegit** [notifiers/notifiers#236]
For me, I need proxy to use telegram and discord, hope this project has proxy support

**Comment by liiight** [comment]
Could you please elaborate on your use case?

**Comment by fakegit** [comment]
telegram is blocked at where i lived, i must send notification to telegram through proxy, i'd like function like the code bellow:
`from notifiers import get_notifier
telegram = get_notifier('telegram')
telegram.notify(message='Hi!', token='TOKEN', chat_id=1234, proxy='http://127.0.0.1:1080')`

**Comment by liiight** [comment]
Yeah, this is very doable. Currently the base URL is a class attribute,  but making it passable should be fairly straightforward

**Comment by liiight** [comment]
Since notifiers uses requests, seems like there's a workaround for this: 
```bash
$ export HTTP_PROXY="http://10.10.1.10:3128"
$ export HTTPS_PROXY="http://10.10.1.10:1080"

$ python
>>> import requests
>>> requests.get('http://example.org')
```
Could you please try this to verify that my implementation idea is valid?
